src/main/kotlin/com/terraformation/backend/tracking/model/PlantingSeasonModel.kt

```diff
@@ -12,3 +12,35 @@ data class ExistingPlantingSeasonModel(
     val startDate: LocalDate,
     val startTime: Instant,
 )
+
+data class UpdatedPlantingSeasonModel(
+    val endDate: LocalDate,
+    val id: PlantingSeasonId? = null,
+    val startDate: LocalDate,
+) {
+  constructor(
+      existingModel: ExistingPlantingSeasonModel
+  ) : this(
+      endDate = existingModel.endDate,
+      id = existingModel.id,
+      startDate = existingModel.startDate,
+  )
+
+  fun validate(currentDate: LocalDate) {
+    if (endDate < startDate.plusDays(28)) {
```

Comment:
do we want another check for endDate <= startDate? (i understand there's a db schema constraint anyways)

Reply:
You mean to return a different error? The existing check should catch that case (if end date is before start date, then it'll also be before start date + 28 days).

Reply:
Of course, my mistake. This is clear, thanks 🙏